thewalrus/_hafnian.py

```diff
@@ -180,25 +181,25 @@ def find_kept_edges(j, reps):  # pragma: no cover
 
 
 @numba.jit(nopython=True, cache=True)
-def f(E, n):  # pragma: no cover
+def f(A, n):  # pragma: no cover
     """Evaluate the polynomial coefficients of the function in the eigenvalue-trace formula.
 
     Args:
-        E (array): eigenvalues of ``AX``
+        A (2d array): matrix
         n (int): number of polynomial coefficients to compute
 
     Returns:
         array: polynomial coefficients
     """
-    E_k = E.copy()
     # Compute combinations in O(n^2log n) time
     # code translated from thewalrus matlab script
     count = 0
     comb = np.zeros((2, n // 2 + 1), dtype=np.complex128)
     comb[0, 0] = 1
+    powtrace = charpoly.powertrace(A, n // 2 + 1)
     for i in range(1, n // 2 + 1):
-        factor = E_k.sum() / (2 * i)
-        E_k *= E
+        # Maybe try powtrace[i-1] instead
```

Comment:
This comment should be removes @dleclerc33 @benjaminlanthier @JQZ1111 @brandonpolymtl

Reply:
```suggestion

```